imgcreate/dnfinst.py

```diff
@@ -195,6 +204,69 @@ def _varSubstitute(option):
         self.repos.add(repo)
         return repo
 
+    def addRepositoryFromConfigFile(self, repo):
+        """
+        Import repository configuration from a DNF repo file instead of
+        from kickstart. If repo is a directory, all *.repo files in that
+        directory are imported.
+        """
+
+        def _to_abs_paths(paths):
+            # replace filenames (relative to repo_dir) with absolute URLs
+            for path in map(str, paths):
+                path = os.path.expanduser(path)
+                absurl = urljoin(
+                    'file://%s/' % os.path.abspath(repo_dir),
+                    path
+                )
+                if not absurl.startswith('file:'):
+                    raise CreatorError(
+                        "livecd-creator does not permit the network URL "
+                        "\"%s\" for GPG keys. Please download this file, "
+                        "manually verify its entire fingerprint, and save "
+                        "it locally." % absurl
+                    )
```

Comment:
This is somewhat opinionated on my part… but since we don't allow the user to confirm keys like interactive dnf does, we also have no business fetching them over the network during an image build.

Reply:
I disagree with this, since I think it's reasonable for keys to be fetched of the network. Other image build tools are perfectly fine with that, and some distributions do not provide local keys for everything. From a workflow perspective, I _personally_ prefer the keys to be local, but this is too much to ask for a lot of people.

Reply:
Fixed in patch v2, 4e68e05426421.